File: identity-service/src/main/java/com/stancloud/identityservice/controller/AuthController.java
package com.stancloud.identityservice.controller;

import com.stancloud.identityservice.dto.UserDto;
import com.stancloud.identityservice.dto.UserResponseDto;
import com.stancloud.identityservice.service.AuthService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/auth")
public class AuthController {
  @Autowired
  private AuthService authService;

  @PostMapping("/register")
  public UserResponseDto register(@RequestBody UserDto userDto) {
    return authService.saveUser(userDto);
  }

  @PostMapping("/login")
  public String login(@RequestBody UserDto userDto) {
    return authService.login(userDto);
  }

  @GetMapping("/validate/{token}")
  public String validateToken(@RequestParam String token) {
    authService.validateToken(token);
    return "Token is valid";
  }



}

File: authentication-service/src/main/kotlin/com/stancloud/authenticationservice/auth/AuthController.kt
package com.stancloud.authenticationservice.auth

import com.stancloud.authenticationservice.dto.LoginRequest
import com.stancloud.authenticationservice.dto.LoginResponse
import com.stancloud.authenticationservice.dto.SignupRequest
import com.stancloud.authenticationservice.dto.SignupResponse
import com.stancloud.authenticationservice.response.ResponseHandler
import com.stancloud.authenticationservice.response.ResponseTemplate
import jakarta.validation.Valid
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/auth")
class authController(
  private val authService: AuthService,
) {

  @PostMapping("/signup")
  fun signup(
    @Valid @RequestBody request: SignupRequest,
  ): SignupResponse {
    return authService.signup(request)
  }

  @PostMapping("/login")
  fun login(
    @RequestBody request: LoginRequest,
  ): ResponseTemplate<String> {
    var response: LoginResponse = authService.login(request)
    return ResponseHandler.generateOkResponse(response.token)
  }

//  @PostMapping("/refresh")
//  fun refresh(
//    @RequestBody request: RefreshRequest,
//  ): RefreshResponse {
//    return authService.refresh(request)
//  }
}